src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventHandler.cs

```diff
@@ -55,15 +62,69 @@ public void Close()
         /// <inheritDoc />
         public void ProcessRequests()
         {
-            // todo : implement this while doing integration with test execution process.
-        }
+            var isSessionEnd = false;
 
-        /// <summary>
-        /// The dispose.
-        /// </summary>
-        public void Dispose()
-        {
-            this.communicationManager?.StopServer();
+            do
+            {
+                var message = this.communicationManager.ReceiveMessage();
+                switch (message.MessageType)
+                {
+                    case MessageType.BeforeTestCaseStart:
+                        if (EqtTrace.IsInfoEnabled)
+                        {
+                            EqtTrace.Info("DataCollectionTestCaseEventHandler : Test case starting.");
```

Comment:
nit: good to check if we are using a space before `:` in log messages for consistency.